admin-dev/themes/new-theme/js/pages/import/index.js

```diff
@@ -0,0 +1,181 @@
+    let $alert = $('.js-entity-alert');
+
+    if ([entityCategories, entityProducts].includes(selectedEntity)) {
+      $alert.show()
+    } else {
+      $alert.hide();
+    }
+  }
+
+  /**
+   * Toggle available options for selected entity
+   *
+   * @param {int} selectedEntity
+   * @param {string} entityName
+   */
+  function toggleFields(selectedEntity, entityName) {
+    const $truncateFormGroup = $('.js-truncate-form-group');
+    const $matchRefFormGroup = $('.js-match-ref-form-group');
+    const $regenerateFormGroup = $('.js-regenerate-form-group');
+    const $forceIdsFormGroup = $('.js-force-ids-form-group');
+    const $entityNamePlaceholder = $('.js-entity-name');
+
+    if (entityStoreContacts === selectedEntity) {
+      $truncateFormGroup.hide();
+    } else {
+      $truncateFormGroup.show();
+    }
+
+    if ([entityProducts, entityCombinations].includes(selectedEntity)) {
+      $matchRefFormGroup.show();
+    } else {
+      $matchRefFormGroup.hide();
+    }
+
+    if ([entityCategories,
+        entityProducts,
+        entityBrands,
+        entitySuppliers,
+        entityStoreContacts
+      ].includes(selectedEntity)
+    ) {
+      $regenerateFormGroup.show();
+    } else {
+      $regenerateFormGroup.hide();
+    }
+
+    if ([entityCategories,
+        entityProducts,
+        entityCustomers,
+        entityAddresses,
+        entityBrands,
+        entitySuppliers,
+        entityStoreContacts,
+        entityAlias
+      ].includes(selectedEntity)
+    ) {
+      $forceIdsFormGroup.show();
+    } else {
+      $forceIdsFormGroup.hide();
+    }
+
+    $entityNamePlaceholder.html(entityName);
+  }
+
+  /**
+   * Load available fields for given entity
+   *
+   * @param {int} entity
+   */
+  function loadAvailableFields(entity) {
+    $.ajax({
+      url: '../../../ajax.php',
+      data: {
+        getAvailableFields:1,
+        entity: entity
+      },
+      dataType: 'json',
+    }).then(response => {
+      let fields = '';
+      let $availableFields = $('.js-available-fields');
+      $availableFields.empty();
+
+      for (let i = 0; i < response.length; i++) {
+        fields += response[i].field;
+      }
+
+      $availableFields.html(fields);
+      $availableFields.find('[data-toggle="popover"]').popover();
+    }).catch(error => {
+
+    });
+  }
+
+  /**
+   * Upload selected import file
+   */
+  function uploadFile() {
+    const uplodedFile = $('#file').prop('files')[0];
+
+    const data = new FormData(uplodedFile);
+    data.append('file', uplodedFile);
+
+    const url = $('.js-import-form').data('file-upload-url');
+
+    //@todo: add progress bar when uploading
+
+    $.ajax({
+      type: 'POST',
+      url: url,
+      data: data,
+      cache: false,
+      contentType: false,
+      processData: false,
+    }).then(response => {
+      let filename = response.file.name;
+      $('.js-uploaded-file').val(filename);
+    }).catch(error => {
+      //@todo: display error to admin?
```

Comment:
We have growl to display error javascript messages

```js
$.growl.error({message: "This is an error."});
```

But I guess we have to include the dependency and the template /c @Quetzacoalt91 ?